api/envoy/extensions/filters/network/thrift_proxy/router/v3/router.proto

```diff
@@ -18,4 +20,7 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 message Router {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.config.filter.thrift.router.v2alpha1.Router";
+
+  // Close downstream connection in case of routing or upstream connection problem.
+  google.protobuf.BoolValue close_downstream_on_upstream_error = 1;
```

Comment:
If we avoid closing a connection safely, then should we always do it?

Reply:
I think we can switch to on by default later on, once this has been used for a bit.

Reply:
Oh nvm, we are defaulting to always doing it:

```
  bool close_downstream_on_error =
      PROTOBUF_GET_WRAPPED_OR_DEFAULT(proto_config, close_downstream_on_upstream_error, true);
      ```

Reply:
> If we can avoid closing a connection safely, then should we always do it?

I just added this flag to ensure backward compatibility. I don't know, maybe someone has built on the close.
Be careful, the `true` here means the original behavor, so close the connection.